Readme.md

```diff
@@ -129,13 +129,276 @@ with keys corresponding to the options described in [CLI Arguments](#cliargs):
+function.
+
+For most parsers, the return value should be a log entry object that will be used by
+the builders to construct the output line.
+
+In some cases, whether for performance or other reasons, it makes sense to return the
+result of a parser directly without performing any additional processing. The process
+can be short-circuited by calling the `stop` method of the `state` argument. No other
+parsers will be used, and the build stage will be skipped as well. The value returned
+by the parser that calls the `stop` function will be returned as the final formatted
+output.
+
+#### building
+
+The `build` function is a simple functions that prepares the final formatted line.
+It receives two parameters: `lineParts` and `context`.
+* The `lineParts` array contains the ordered list of strings that will eventually be joined
+  after all the build functions have been executed.
+* The `context` object is the same `context` object passed to the parsers, with one
+  exception: the final output produced by the parsers is added to the object as the `log`
+  property.
+
+  Also, when the built-in prettifying log processors are used, a number of prettified
+  strings are added to the `context.prettified` property: `prettifiedLevel`,
+  `prettifiedMessage`, `prettifiedMetadata`, and `prettifiedTime`.
+
+The builder function does not return a value. To change the output, modify the
+`lineParts` array.
+
+Note that if a parser short-circuits the parsing process, the builders will not be
+executed at all.
+
+#### examples
+
+The following are a number of examples of parsers, builders, and combined log processors:
+
+* **parser** - Perform a transformation on the input message:
+
+  ```js
+  function (input) {
+    input.msg = input.msg.toLowerCase()
+    return input
+  }
+  ```
+
+* **parser** - Short-circuit the parsing process and prevent subsequent log parsers from
+  being executed:
+
+  ```js
+  (input, context, state) => {
+    state.stop() // short-circuit the log parsing process
+    return input
+  }
+  ```
+
+* **parser** - Prepare a prettified string:
+
+  ```js
+  {
+    parse(input, { colorizer, inlineSection ) {
+      if(inlineSection && 'section' in input) {
+        context.prettified.section = `[${colorizer(input.section)}]`
+      }
+      return input
+    }
+  }
+  ```
+
+* **builder** - Add a value to the array:
+
+  ```js
+  function (lineParts) {
+    lineParts.push('NEW VALUE')
+  }
+  ```
+
+* **builder** - The following is an example of a more complex builder function, based
+  on one of the built-in builders:
+
+  ```js
+  {
+    build (lineParts, { prettified }) => {
+      const { prettifiedTime } = prettified
+      if (prettifiedTime) {
+        if (lineParts.length > 0) {
+          lineParts.push(' ')
+        }
+        lineParts.push(prettifiedTime)
+      }
+    }
+  }
+  ```
+
+* **parser** and **builder** - Parse a log entry and then use the result to build the
+  output line:
```

Comment:
This sentence makes it sound like the `build` function will be used _immediately_ after the provided `parse` function is used. Is that the case? Or is the `build` function used during the build process?

Reply:
No, you're right. The `build` function is used during the build stage. I've updated the Readme to clarify.